app/components/travis-switch.js

```diff
@@ -1,28 +1,28 @@
-import { next } from '@ember/runloop';
 import Component from '@ember/component';
-import { or } from 'ember-decorators/object/computed';
+import { computed } from 'ember-decorators/object';
 
 export default Component.extend({
   tagName: 'button',
-  classNames: ['travis-switch', 'switch'],
-  classNameBindings: ['_active:active'],
+  classNames: ['travis-switch', 'switch', 'inline-block'],
+  classNameBindings: ['active:active', 'disabled:disabled'],
+  attributeBindings: ['aria-checked', 'role', 'type'],
 
-  attributeBindings: ['type'],
   type: 'button',
+  role: 'switch',
+  description: '',
+  active: false,
+  disabled: false,
 
-  @or('target.active', 'active') _active: null,
+  @computed('active')
+  'aria-checked'(active) {
+    return active ? 'true' : 'false';
+  },
+
+  onToggle() {},
```

Comment:
In terms of developer ergonomics, what do you think of adding something like this here?

```
onToggle() {
  throw new Error('Must supply an implementation for onToggle');
}
```

It might slightly ease the work of some future person using the component 😀

Reply:
Yeah, I agree, totally makes sense 👍 Will push a fix shortly

Reply:
@backspace I added and [abstractMethod util](https://github.com/travis-ci/travis-web/pull/1756/files#diff-77c6b499af98e4d8e6c7c30d046fa488) for such cases